Do not show the titles in the admin proposals page if there isn't any

andreslucena opened the pull request:
#### :tophat: What? Why?

When seeing the page for a proposal in the admin panel, it's showing some sections that don't have any content ("Endorsers" and "Related meetings"). 

This PR fixes it by only showing them if they have content

#### :pushpin: Related Issues
*Link your PR to an issue*
- Related to #?
- Fixes #?

#### Testing

1. Sign in as admin
2. Go to a proposal page in the admin
3. (without the patch) see that you have these title sections even though there isn't any Endorsement nor Related meeting
4. (with the patch):
4.1. See that you don't have these titles if there isn't any content
4.2. See that you have them if there is content

### :camera: Screenshots

#### Before
![Screenshot of this page](https://github.com/decidim/decidim/assets/717367/73922272-020f-4671-8704-2a8c2418beb6)

#### After
![Screenshot of this page](https://github.com/decidim/decidim/assets/717367/f922dbbb-1ffa-46eb-8e2c-cd6f67f8111b)

:hearts: Thank you!

andreslucena commented:
This bug is happening in v0.27 so it needs to be backported:

![Screenshot of this page in v0.27 with the bug](https://github.com/decidim/decidim/assets/717367/4f251961-ecd0-4cc4-87c9-cdab80fc0bae)

alecslupu commented on decidim-proposals/app/views/decidim/proposals/admin/proposals/show.html.erb:
```diff
@@ -49,32 +49,34 @@
-                </li>
-              <% end %>
-            </ul>
+        <% presented_endorsers = endorsers_presenters_for(proposal).to_a %>
```
```suggestion
        <% presented_endorsers = endorsers_presenters_for(proposal).first(6).to_a %>
```

If there is a very long list of endorsers ( let's think tens or hundreds), this could generate a memory bloat.

alecslupu commented on decidim-proposals/spec/system/admin/view_proposal_details_from_admin_spec.rb:
```diff
@@ -113,49 +113,60 @@
-        6.times.collect do
-          create(:endorsement, resource: proposal, author: build(:user, organization:))
+      context "with more than 5 endorsements" do
+        let!(:endorsements) do
+          6.times.collect do
+            create(:endorsement, resource: proposal, author: build(:user, organization:))
+          end
```
Why not simpler ?
```suggestion
        let!(:endorsements) do
            create_list(:endorsement, 6, resource: proposal, author: build(:user, organization:))
```

alecslupu commented on decidim-proposals/spec/system/admin/view_proposal_details_from_admin_spec.rb:
```diff
@@ -113,49 +113,60 @@
-      go_to_admin_proposal_page(proposal)
+    context "when there are endorsements" do
+      let!(:endorsements) do
+        2.times.collect do
+          create(:endorsement, resource: proposal, author: build(:user, organization:))
+        end
```
Why not simpler ?

```suggestion
          create_list(:endorsement, 2, resource: proposal, author: build(:user, organization:))

```

alecslupu commented on decidim-proposals/app/views/decidim/proposals/admin/proposals/show.html.erb:
```diff
@@ -49,32 +49,34 @@
-            </ul>
+        <% presented_endorsers = endorsers_presenters_for(proposal).to_a %>
+        <% if presented_endorsers.any? %>
+          <div class="card-section">
```
I am thinking maybe is worth extracting to a partial ?

andreslucena replied:
I agree, but let's do it in another PR as a refactor. As it is now I think would be easier to backport to v0.27 and v0.26

alecslupu commented on decidim-proposals/app/views/decidim/proposals/admin/proposals/show.html.erb:
```diff
@@ -105,19 +107,22 @@
-            </ul>
+        <% proposal_meetings = proposal.linked_resources(:meetings, "proposals_from_meeting") %>
+        <% if proposal_meetings.any? %>
+          <div class="card-section">
```
I am thinking maybe is worth extracting to a partial ?

andreslucena replied:
Same here, I propose to do it as another PR

alecslupu replied:
works for me.

alecslupu reviewed (changes requested):
The fix works as advertised. There are some style and performance concerns.

alecslupu reviewed (approved):
:+1:

alecslupu reviewed (changes requested):
apparently the proposals specs are failing ... 

i have found the answer for "why not simpler?" ... " 'cause it is failing". 

Let's revert to the the x.times syntax ... Thanks

andreslucena commented:
> apparently the proposals specs are failing ...
> 
> i have found the answer for "why not simpler?" ... " 'cause it is failing".
> 
> Let's revert to the the x.times syntax ... Thanks

I've reverted it and also added a comment to make it clear that using the x.times syntax is an intentional thing

alecslupu reviewed (approved):
:+1:
